tests/ConfigurationSchemaGenerator.Tests/GeneratorTests.cs

```diff
@@ -1297,6 +1436,122 @@ public class TestSettings
             """);
     }
 
+    [Fact]
+    public void OverwritesLogCategoriesWithDifferentCasing()
```

Comment:
I think we should be consistent when we have conflicts. Either the first one should win, or the last one should win. I don't think we should have different behavior depending on where the conflict exists.

Reply:
Ok, I've made the last one win in all cases and clarified the test names.
And updated the PR description to reflect these changes.